packages/ui/src/components/Visualization/Canvas/FormV2/fields/ExpressionField/ExpressionField.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { CamelCatalogService, CatalogKind } from '../../../../../../models';
+import { getFirstCatalogMap } from '../../../../../../stubs/test-load-catalog';
+import { ModelContextProvider } from '../../providers/ModelProvider';
+import { SchemaDefinitionsProvider } from '../../providers/SchemaDefinitionsProvider';
+import { FormComponentFactoryProvider } from '../../providers/FormComponentFactoryProvider';
+import { KaotoFormPageObject } from '../../testing/KaotoFormPageObject';
+import { setHeaderExpressionSchema } from '../../../../../../stubs/expression-definition-schema';
+
+describe('ExpressionField', () => {
+  beforeEach(async () => {
+    const catalogsMap = await getFirstCatalogMap(catalogLibrary as CatalogLibrary);
+    CamelCatalogService.setCatalogKey(CatalogKind.Language, catalogsMap.languageCatalog);
+  });
+
+  it('renders empty expression field with schema', () => {
+    const { container } = render(
+      <ModelContextProvider model={{ id: 'setHeader-1361' }} onPropertyChange={jest.fn()}>
+        <SchemaProvider schema={setHeaderExpressionSchema}>
+          <ExpressionField propName={ROOT_PATH} required={true} />
+        </SchemaProvider>
+      </ModelContextProvider>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders expression field with selection', () => {
+    const { container } = render(
+      <FormComponentFactoryProvider>
+        <ModelContextProvider
+          model={{
+            id: 'setHeader-1891',
+            expression: {
+              simple: {},
+            },
+          }}
+          onPropertyChange={jest.fn()}
+        >
+          <SchemaDefinitionsProvider schema={setHeaderExpressionSchema} omitFields={[]}>
+            <SchemaProvider schema={setHeaderExpressionSchema}>
+              <ExpressionField propName={ROOT_PATH} required={true} />
+            </SchemaProvider>
+          </SchemaDefinitionsProvider>
+        </ModelContextProvider>
+      </FormComponentFactoryProvider>,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should be able to change the selection', async () => {
+    render(
+      <FormComponentFactoryProvider>
+        <ModelContextProvider
+          model={{
+            id: 'setHeader-1891',
+            expression: {
+              simple: {},
+            },
+          }}
+          onPropertyChange={jest.fn()}
+        >
+          <SchemaDefinitionsProvider schema={setHeaderExpressionSchema} omitFields={[]}>
+            <SchemaProvider schema={setHeaderExpressionSchema}>
+              <ExpressionField propName={ROOT_PATH} required={true} />
+            </SchemaProvider>
+          </SchemaDefinitionsProvider>
+        </ModelContextProvider>
+      </FormComponentFactoryProvider>,
+    );
+
+    const formPageObject = new KaotoFormPageObject(screen, act);
+    await formPageObject.toggleExpressionFieldForProperty(ROOT_PATH);
+    await formPageObject.selectTypeaheadItem(ROOT_PATH, 'csimple');
```

Comment:
Can we please add a verification about `csimple` being selected? Perhaps we could get the input field and check its value